plugins/linux-capture/pipewire.c

```diff
@@ -655,6 +698,18 @@ static void play_pipewire_stream(obs_pipewire_data *obs_pw)
 	pw_core_add_listener(obs_pw->core, &obs_pw->core_listener, &core_events,
 			     obs_pw);
 
+	// Dispatch to receive the info core event
+	while (obs_pw->server_version.major == 0 &&
+	       obs_pw->server_version.minor == 0 &&
+	       obs_pw->server_version.micro == 0) {
```

Comment:
This won't work since `info->version` is only available in the `on_core_info_cb`, we we don't want to duplicate it as a global.
But waiting is a good idea. Used the `server_version_sync` to check if the core events were triggered by us.

Reply:
This diff seems to work:

```diff
diff --git a/plugins/linux-capture/pipewire.c b/plugins/linux-capture/pipewire.c
index 6ca3ec232..1ccfcc750 100644
--- a/plugins/linux-capture/pipewire.c
+++ b/plugins/linux-capture/pipewire.c
@@ -104,6 +104,7 @@ struct _obs_pipewire_data {
 
        enum obs_pw_capture_type capture_type;
        struct obs_video_info video_info;
+       int server_version_sync;
        bool negotiated;
 };
 
@@ -616,12 +617,21 @@ static void on_core_done_cb(void *user_data, uint32_t id, int seq)
 
        if (id == PW_ID_CORE)
                pw_thread_loop_signal(obs_pw->thread_loop, FALSE);
+
+       if (obs_pw->server_version_sync = seq)
+               pw_thread_loop_signal(obs_pw->thread_loop, false);
+}
+
+static void on_core_info_cb(void *user_data, const struct pw_core_info *info)
+{
+       // Add the update_version() call here
+       blog(LOG_INFO, "server version: %s", info->version);
 }
 
 static const struct pw_core_events core_events = {
        PW_VERSION_CORE_EVENTS,
        .done = on_core_done_cb,
        .error = on_core_error_cb,
+       .info = on_core_info_cb,
 };
 
 static void play_pipewire_stream(obs_pipewire_data *obs_pw)
@@ -655,6 +665,9 @@ static void play_pipewire_stream(obs_pipewire_data *obs_pw)
        pw_core_add_listener(obs_pw->core, &obs_pw->core_listener, &core_events,
                             obs_pw);
 
+       obs_pw->server_version_sync = pw_core_sync(obs_pw->core, PW_ID_CORE, obs_pw->server_version_sync);
+       pw_thread_loop_wait(obs_pw->thread_loop);
+
        /* Stream */
        obs_pw->stream = pw_stream_new(
                obs_pw->core, "OBS Studio",
```